### PR TITLE
Generate bitcode for plugin frameworks for flutter build ios-framework

### LINKED ISSUE #48092: Add2App [Option A&B] with plugins failed to build because plugins was built without full bitcode
This is a next step of an issue described here #47930 and all preparation steps is the same.
But now before building `.xcframework` just add to `my_flutter` module some plugin dependencies. 
I added most popular _shared_preferences_ plugin to `pubspec.yaml` and added example code to `main.dart` file.
Next
```
flutter build ios-framework --output=../host_ios/embed/ --xcframework --no-universal
```
produce 
```
App.framework
Flutter.xcframework
FlutterPluginRegistrant.xcframework
shared_preferences.xcframework
```
Applying them to xcode pbxproj -> build -> error

> ld: bitcode bundle could not be generated because '/Users/n/Library/Developer/Xcode/DerivedData/host_ios-bnbwehcmdjzqgydxspnsnfifvdyh/Build/Intermediates.noindex/ArchiveIntermediates/host_ios/BuildProductsPath/Release-iphoneos/shared_preferences.framework/shared_preferences' was built without full bitcode. All frameworks and dylibs for bitcode must be generated from Xcode Archive or Install build file '/Users/n/Library/Developer/Xcode/DerivedData/host_ios-bnbwehcmdjzqgydxspnsnfifvdyh/Build/Intermediates.noindex/ArchiveIntermediates/host_ios/BuildProductsPath/Release-iphoneos/shared_preferences.framework/shared_preferences' for architecture arm64

I checked this situation with another plugin - _path_provider_ - result the same.

If I try to disable *bitcode* I get another error:
headerr
with this logs:
[IDEDistribution.critical.log](https://github.com/flutter/flutter/files/4017571/IDEDistribution.critical.log)
[IDEDistribution.standard.log](https://github.com/flutter/flutter/files/4017572/IDEDistribution.standard.log)
[IDEDistribution.verbose.log](https://github.com/flutter/flutter/files/4017573/IDEDistribution.verbose.log)

### PR DESCRIPTION
**Pull request by jmagman**
## Description

1. The plugin frameworks didn't have bitcode embedded because they were being built, not archived.  Add `BITCODE_GENERATION_MODE` when building to force marker or actual bitcode, depending on configuration.

2. While building the Flutter plugins, they were linking on the last version of Flutter.framework copied from `flutter build`/`run`.  So if you ran `flutter build ios --debug` then `flutter build ios-framework`, the Release plugins would not compile with bitcode since they were linking on the debug version of Flutter.framework, which does not contain bitcode.

## Related Issues

Fixes https://github.com/flutter/flutter/issues/48092

## Tests

- Added bitcode checks to build_ios_framework_module_test integration test.

## Checklist
- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change
- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*

### CODE REVIEW

File: packages/flutter_tools/lib/src/commands/build_ios_framework.dart [hunk 1]
```diff
@@ -266,10 +266,7 @@ end
   }
 
   Future<void> _produceFlutterFramework(
-    Directory outputDirectory,
     BuildMode mode,
-    Directory iPhoneBuildOutput,
-    Directory simulatorBuildOutput,
```
Comment (jmagman) [inline packages/flutter_tools/lib/src/commands/build_ios_framework.dart]: These params were unused.

File: dev/devicelab/lib/framework/ios.dart [hunk 2]
```diff
@@ -54,6 +54,48 @@ Future<String> dylibSymbols(String pathToDylib) {
...
+  //  size 0x0042b633
+  //  offset 3932160
+  //  ...
+  final List<String> lines = LineSplitter.split(loadCommands).toList();
+  lines.asMap().forEach((int index, String line) {
+    if (line.contains('segname __LLVM') && lines.length - index - 1 > 3) {
+      final String emptyBitcodeMarker = lines
+        .skip(index - 1)
+        .take(3)
+        .firstWhere(
```
Comment (christopherfujino) [inline dev/devicelab/lib/framework/ios.dart]: this is fun

**Review by christopherfujino — COMMENTED** [review]
Are we testing the flow of:
```
flutter build ios --debug
flutter build ios-framework # verify bitcode
```
?

File: dev/devicelab/lib/framework/ios.dart (hunk 2 above)
Comment (jmagman) [reply dev/devicelab/lib/framework/ios.dart]: Thanks I had "fun" writing it.

**Review by jmagman — COMMENTED** [review]
> Are we testing the flow of:
> 
> ```
> flutter build ios --debug
> flutter build ios-framework # verify bitcode
> ```
> 
> ?

You mean to prove the Release plugins are linking against the Release version of Flutter.framework?  No, but that's a good idea (though will add a good chunk of time to this test).

**Review by christopherfujino — APPROVED** [review]
LGTM

**Pull request merged**